components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIDefinitionFromSwagger20.java

```diff
@@ -412,6 +422,16 @@ public String generateSwaggerFromResources(API.APIBuilder api) {
                 operation.setConsumes(consumesList);
             }
             operation.addResponse("200", getDefaultResponse());
+            if (!uriTemplate.getAuthType().equals(APIMgtConstants.AUTH_NO_AUTHENTICATION) &&
```

Comment:
Better to check the condition other way to avoid NPE. Eg: !APIMgtConstants.AUTH_NO_AUTHENTICATION.equals(uriTemplate.getAuthType())

Reply:
ack